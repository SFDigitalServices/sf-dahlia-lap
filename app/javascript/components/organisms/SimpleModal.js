import React from 'react'
import classNames from 'classnames'

import Modal from './Modal'

const SimpleModalBody = ({
  type,
  header,
  isOpen,
  primary,
  secondary,
  alert,
  invert,
  handleClose,
  onPrimaryClick,
  onSecondaryClick,
  children
}) => {
  const primaryButtonClassName = classNames({
    button: true,
    primary: !type || type === 'status',
    'alert-fill': type === 'alert'
  })

  return (
    <Modal.Body hidden={isOpen} handleClose={handleClose}>
      <Modal.Header title={header} />
      {alert && <Modal.Alert {...alert} />}
      <Modal.Content>{children}</Modal.Content>
      <Modal.Footer>
        <div className='modal-button_item modal-button_primary'>
          <button className={primaryButtonClassName} onClick={onPrimaryClick}>
            {primary}
          </button>
        </div>
        <div className='modal-button_item modal-button_secondary'>
          <button className='button no-border' onClick={onSecondaryClick}>
            {secondary}
          </button>
        </div>
      </Modal.Footer>
    </Modal.Body>
  )
}

const SimpleModal = ({ children, isOpen, handleClose, ...SimpleModalBodyProps }) => {
  return (
    <Modal isOpen={isOpen} handleClose={handleClose}>
      <SimpleModalBody handleClose={handleClose} {...SimpleModalBodyProps}>
        {children}
      </SimpleModalBody>
    </Modal>
  )
}

export default SimpleModal
