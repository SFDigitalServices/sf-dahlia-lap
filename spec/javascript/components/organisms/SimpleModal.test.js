/* global mount */
import React from 'react'
// import sinon from 'sinon'

import SimpleModal from 'components/organisms/SimpleModal'

describe('SimpleModal', () => {
  const isOpen = true
  const onCloseClick = jest.fn()
  const onPrimaryClick = jest.fn()
  const onSecondaryClick = jest.fn()

  test('it should render status type successfully', () => {
    const wrapper = mount(
      <SimpleModal
        header='Update Status'
        primary='update'
        secondary='cancel'
        isOpen={isOpen}
        handleClose={onCloseClick}
        onPrimaryClick={onPrimaryClick}
        onSecondaryClick={onSecondaryClick}
        type='status'
        alert={{
          title: "This change will affect this application's preferences",
          subtitle: 'This application would no longer be eligible for Live Work Preference',
          message:
            'Note, you will have the opportunity to grant another household member this preference',
          invert: false
        }}
      >
        <div>content</div>
      </SimpleModal>
    )

    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ type: 'alert' })

    expect(wrapper).toMatchSnapshot()
  })

  test('it should call event handlers', () => {
    const wrapper = mount(
      <SimpleModal
        header='Update Status'
        primary='update'
        secondary='cancel'
        isOpen={isOpen}
        handleClose={onCloseClick}
        onPrimaryClick={onPrimaryClick}
        onSecondaryClick={onSecondaryClick}
        type='status'
      >
        <div>content</div>
      </SimpleModal>
    )

    wrapper.find('.modal-button_primary button').simulate('click')
    wrapper.find('.modal-button_secondary button').simulate('click')
    wrapper.find('.close-reveal-modal').simulate('click')

    expect(onPrimaryClick.mock.calls).toHaveLength(1)
    expect(onSecondaryClick.mock.calls).toHaveLength(1)
    expect(onCloseClick.mock.calls).toHaveLength(1)
  })
})
