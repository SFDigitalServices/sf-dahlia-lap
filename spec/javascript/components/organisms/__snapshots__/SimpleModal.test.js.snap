// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleModal it should render status type successfully 1`] = `
<SimpleModal
  alert={
    Object {
      "invert": false,
      "message": "Note, you will have the opportunity to grant another household member this preference",
      "subtitle": "This application would no longer be eligible for Live Work Preference",
      "title": "This change will affect this application's preferences",
    }
  }
  handleClose={[MockFunction]}
  header="Update Status"
  isOpen={true}
  onPrimaryClick={[MockFunction]}
  onSecondaryClick={[MockFunction]}
  primary="update"
  secondary="cancel"
  type="status"
>
  <Modal
    handleClose={[MockFunction]}
    isOpen={true}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      isOpen={true}
      onRequestClose={[MockFunction]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "border": "0",
            "borderRadius": "4px",
            "bottom": "auto",
            "left": "50%",
            "minHeight": "10rem",
            "opacity": "1",
            "overflow": "visible",
            "padding": "0px",
            "position": "fixed",
            "right": "auto",
            "top": "50%",
            "transform": "translate(-50%,-50%)",
            "width": "600px",
          },
          "overlay": Object {
            "backgroundColor": "rgba(0,0,0,0.3)",
          },
        }
      }
    >
      <Portal
        containerInfo={
          <div
            class="ReactModalPortal"
          >
            <div
              class="ReactModal__Overlay ReactModal__Overlay--after-open"
              style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(0, 0, 0, 0.3);"
            >
              <div
                class="ReactModal__Content ReactModal__Content--after-open"
                role="dialog"
                style="position: fixed; top: 50%; left: 50%; border: 0px; background: rgb(255, 255, 255); overflow: visible; border-radius: 4px; outline: none; padding: 0px; opacity: 1; min-height: 10rem; transform: translate(-50%,-50%); width: 600px;"
                tabindex="-1"
              >
                <div
                  aria-labelledby="modalTitle"
                  role="dialog"
                >
                  <header
                    class="modal-inner margin-top"
                  >
                    <h1
                      class="modal-title t-gamma no-margin"
                    >
                      Update Status
                    </h1>
                  </header>
                  <div
                    class="alert-box-and-notice"
                  >
                    <div
                      class="alert-box alert no-margin"
                      data-alert="true"
                    >
                      <span
                        class="alert-icon ui-icon ui-medium"
                      >
                        <svg>
                          <use
                            xlink:href="#i-warning"
                          />
                        </svg>
                      </span>
                      <p
                        class="alert-body"
                      >
                        This change will affect this application's preferences
                      </p>
                      <button
                        class="button button-link close text"
                        type="button"
                      >
                        Close
                      </button>
                    </div>
                    <div
                      class="alert-notice alert "
                    >
                      <p
                        class="t-tiny c-alert margin-bottom"
                      >
                        This application would no longer be eligible for Live Work Preference
                      </p>
                      <p
                        class="t-tiny c-steel margin-bottom"
                      >
                        Note, you will have the opportunity to grant another household member this preference
                      </p>
                    </div>
                  </div>
                  <section
                    class="modal-inner"
                  >
                    <div>
                      content
                    </div>
                  </section>
                  <footer
                    class="modal-footer bg-dust"
                  >
                    <div
                      class="modal-button-group row"
                    >
                      <div
                        class="modal-button_item modal-button_primary"
                      >
                        <button
                          class="button primary"
                        >
                          update
                        </button>
                      </div>
                      <div
                        class="modal-button_item modal-button_secondary"
                      >
                        <button
                          class="button no-border"
                        >
                          cancel
                        </button>
                      </div>
                    </div>
                  </footer>
                  <button
                    aria-label="Close modal"
                    class="button button-link close-reveal-modal"
                  >
                    <span
                      class="sr-only"
                    >
                      Close
                    </span>
                    <span
                      class="ui-icon ui-medium i-primary"
                    >
                      <svg>
                        <use
                          xlink:href="#i-close"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        }
      >
        <ModalPortal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          onRequestClose={[MockFunction]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {
                "border": "0",
                "borderRadius": "4px",
                "bottom": "auto",
                "left": "50%",
                "minHeight": "10rem",
                "opacity": "1",
                "overflow": "visible",
                "padding": "0px",
                "position": "fixed",
                "right": "auto",
                "top": "50%",
                "transform": "translate(-50%,-50%)",
                "width": "600px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(0,0,0,0.3)",
              },
            }
          }
        >
          <div
            className="ReactModal__Overlay ReactModal__Overlay--after-open"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.3)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              }
            }
          >
            <div
              className="ReactModal__Content ReactModal__Content--after-open"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              role="dialog"
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "#fff",
                  "border": "0",
                  "borderRadius": "4px",
                  "bottom": "auto",
                  "left": "50%",
                  "minHeight": "10rem",
                  "opacity": "1",
                  "outline": "none",
                  "overflow": "visible",
                  "padding": "0px",
                  "position": "fixed",
                  "right": "auto",
                  "top": "50%",
                  "transform": "translate(-50%,-50%)",
                  "width": "600px",
                }
              }
              tabIndex="-1"
            >
              <SimpleModalBody
                alert={
                  Object {
                    "invert": false,
                    "message": "Note, you will have the opportunity to grant another household member this preference",
                    "subtitle": "This application would no longer be eligible for Live Work Preference",
                    "title": "This change will affect this application's preferences",
                  }
                }
                handleClose={[MockFunction]}
                header="Update Status"
                onPrimaryClick={[MockFunction]}
                onSecondaryClick={[MockFunction]}
                primary="update"
                secondary="cancel"
                type="status"
              >
                <Component
                  handleClose={[MockFunction]}
                >
                  <div
                    aria-labelledby="modalTitle"
                    role="dialog"
                  >
                    <Component
                      title="Update Status"
                    >
                      <header
                        className="modal-inner margin-top"
                      >
                        <h1
                          className="modal-title t-gamma no-margin"
                        >
                          Update Status
                        </h1>
                      </header>
                    </Component>
                    <Component
                      invert={false}
                      message="Note, you will have the opportunity to grant another household member this preference"
                      subtitle="This application would no longer be eligible for Live Work Preference"
                      title="This change will affect this application's preferences"
                    >
                      <Alert
                        invert={false}
                        message="Note, you will have the opportunity to grant another household member this preference"
                        subtitle="This application would no longer be eligible for Live Work Preference"
                        title="This change will affect this application's preferences"
                      >
                        <div
                          className="alert-box-and-notice"
                        >
                          <AlertBox
                            closeType="text"
                            invert={false}
                            message="This change will affect this application's preferences"
                            noMargin={true}
                          >
                            <div
                              className="alert-box alert no-margin"
                              data-alert={true}
                            >
                              <span
                                className="alert-icon ui-icon ui-medium"
                              >
                                <svg>
                                  <use
                                    xlinkHref="#i-warning"
                                  />
                                </svg>
                              </span>
                              <p
                                className="alert-body"
                              >
                                This change will affect this application's preferences
                              </p>
                              <button
                                className="button button-link close text"
                                type="button"
                              >
                                Close
                              </button>
                            </div>
                          </AlertBox>
                          <AlertNotice
                            content="Note, you will have the opportunity to grant another household member this preference"
                            invert={false}
                            title="This application would no longer be eligible for Live Work Preference"
                          >
                            <div
                              className="alert-notice alert "
                            >
                              <p
                                className="t-tiny c-alert margin-bottom"
                              >
                                This application would no longer be eligible for Live Work Preference
                              </p>
                              <p
                                className="t-tiny c-steel margin-bottom"
                              >
                                Note, you will have the opportunity to grant another household member this preference
                              </p>
                            </div>
                          </AlertNotice>
                        </div>
                      </Alert>
                    </Component>
                    <Component>
                      <section
                        className="modal-inner"
                      >
                        <div>
                          content
                        </div>
                      </section>
                    </Component>
                    <Component>
                      <footer
                        className="modal-footer bg-dust"
                      >
                        <div
                          className="modal-button-group row"
                        >
                          <div
                            className="modal-button_item modal-button_primary"
                          >
                            <button
                              className="button primary"
                              onClick={[MockFunction]}
                            >
                              update
                            </button>
                          </div>
                          <div
                            className="modal-button_item modal-button_secondary"
                          >
                            <button
                              className="button no-border"
                              onClick={[MockFunction]}
                            >
                              cancel
                            </button>
                          </div>
                        </div>
                      </footer>
                    </Component>
                    <button
                      aria-label="Close modal"
                      className="button button-link close-reveal-modal"
                      onClick={[MockFunction]}
                    >
                      <span
                        className="sr-only"
                      >
                        Close
                      </span>
                      <span
                        className="ui-icon ui-medium i-primary"
                      >
                        <svg>
                          <use
                            xlinkHref="#i-close"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </Component>
              </SimpleModalBody>
            </div>
          </div>
        </ModalPortal>
      </Portal>
    </Modal>
  </Modal>
</SimpleModal>
`;

exports[`SimpleModal it should render status type successfully 2`] = `
<SimpleModal
  alert={
    Object {
      "invert": false,
      "message": "Note, you will have the opportunity to grant another household member this preference",
      "subtitle": "This application would no longer be eligible for Live Work Preference",
      "title": "This change will affect this application's preferences",
    }
  }
  handleClose={[MockFunction]}
  header="Update Status"
  isOpen={true}
  onPrimaryClick={[MockFunction]}
  onSecondaryClick={[MockFunction]}
  primary="update"
  secondary="cancel"
  type="alert"
>
  <Modal
    handleClose={[MockFunction]}
    isOpen={true}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      isOpen={true}
      onRequestClose={[MockFunction]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "border": "0",
            "borderRadius": "4px",
            "bottom": "auto",
            "left": "50%",
            "minHeight": "10rem",
            "opacity": "1",
            "overflow": "visible",
            "padding": "0px",
            "position": "fixed",
            "right": "auto",
            "top": "50%",
            "transform": "translate(-50%,-50%)",
            "width": "600px",
          },
          "overlay": Object {
            "backgroundColor": "rgba(0,0,0,0.3)",
          },
        }
      }
    >
      <Portal
        containerInfo={
          <div
            class="ReactModalPortal"
          >
            <div
              class="ReactModal__Overlay ReactModal__Overlay--after-open"
              style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(0, 0, 0, 0.3);"
            >
              <div
                class="ReactModal__Content ReactModal__Content--after-open"
                role="dialog"
                style="position: fixed; top: 50%; left: 50%; border: 0px; background: rgb(255, 255, 255); overflow: visible; border-radius: 4px; outline: none; padding: 0px; opacity: 1; min-height: 10rem; transform: translate(-50%,-50%); width: 600px;"
                tabindex="-1"
              >
                <div
                  aria-labelledby="modalTitle"
                  role="dialog"
                >
                  <header
                    class="modal-inner margin-top"
                  >
                    <h1
                      class="modal-title t-gamma no-margin"
                    >
                      Update Status
                    </h1>
                  </header>
                  <div
                    class="alert-box-and-notice"
                  >
                    <div
                      class="alert-box alert no-margin"
                      data-alert="true"
                    >
                      <span
                        class="alert-icon ui-icon ui-medium"
                      >
                        <svg>
                          <use
                            xlink:href="#i-warning"
                          />
                        </svg>
                      </span>
                      <p
                        class="alert-body"
                      >
                        This change will affect this application's preferences
                      </p>
                      <button
                        class="button button-link close text"
                        type="button"
                      >
                        Close
                      </button>
                    </div>
                    <div
                      class="alert-notice alert "
                    >
                      <p
                        class="t-tiny c-alert margin-bottom"
                      >
                        This application would no longer be eligible for Live Work Preference
                      </p>
                      <p
                        class="t-tiny c-steel margin-bottom"
                      >
                        Note, you will have the opportunity to grant another household member this preference
                      </p>
                    </div>
                  </div>
                  <section
                    class="modal-inner"
                  >
                    <div>
                      content
                    </div>
                  </section>
                  <footer
                    class="modal-footer bg-dust"
                  >
                    <div
                      class="modal-button-group row"
                    >
                      <div
                        class="modal-button_item modal-button_primary"
                      >
                        <button
                          class="button alert-fill"
                        >
                          update
                        </button>
                      </div>
                      <div
                        class="modal-button_item modal-button_secondary"
                      >
                        <button
                          class="button no-border"
                        >
                          cancel
                        </button>
                      </div>
                    </div>
                  </footer>
                  <button
                    aria-label="Close modal"
                    class="button button-link close-reveal-modal"
                  >
                    <span
                      class="sr-only"
                    >
                      Close
                    </span>
                    <span
                      class="ui-icon ui-medium i-primary"
                    >
                      <svg>
                        <use
                          xlink:href="#i-close"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        }
      >
        <ModalPortal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          onRequestClose={[MockFunction]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {
                "border": "0",
                "borderRadius": "4px",
                "bottom": "auto",
                "left": "50%",
                "minHeight": "10rem",
                "opacity": "1",
                "overflow": "visible",
                "padding": "0px",
                "position": "fixed",
                "right": "auto",
                "top": "50%",
                "transform": "translate(-50%,-50%)",
                "width": "600px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(0,0,0,0.3)",
              },
            }
          }
        >
          <div
            className="ReactModal__Overlay ReactModal__Overlay--after-open"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.3)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              }
            }
          >
            <div
              className="ReactModal__Content ReactModal__Content--after-open"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              role="dialog"
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "#fff",
                  "border": "0",
                  "borderRadius": "4px",
                  "bottom": "auto",
                  "left": "50%",
                  "minHeight": "10rem",
                  "opacity": "1",
                  "outline": "none",
                  "overflow": "visible",
                  "padding": "0px",
                  "position": "fixed",
                  "right": "auto",
                  "top": "50%",
                  "transform": "translate(-50%,-50%)",
                  "width": "600px",
                }
              }
              tabIndex="-1"
            >
              <SimpleModalBody
                alert={
                  Object {
                    "invert": false,
                    "message": "Note, you will have the opportunity to grant another household member this preference",
                    "subtitle": "This application would no longer be eligible for Live Work Preference",
                    "title": "This change will affect this application's preferences",
                  }
                }
                handleClose={[MockFunction]}
                header="Update Status"
                onPrimaryClick={[MockFunction]}
                onSecondaryClick={[MockFunction]}
                primary="update"
                secondary="cancel"
                type="alert"
              >
                <Component
                  handleClose={[MockFunction]}
                >
                  <div
                    aria-labelledby="modalTitle"
                    role="dialog"
                  >
                    <Component
                      title="Update Status"
                    >
                      <header
                        className="modal-inner margin-top"
                      >
                        <h1
                          className="modal-title t-gamma no-margin"
                        >
                          Update Status
                        </h1>
                      </header>
                    </Component>
                    <Component
                      invert={false}
                      message="Note, you will have the opportunity to grant another household member this preference"
                      subtitle="This application would no longer be eligible for Live Work Preference"
                      title="This change will affect this application's preferences"
                    >
                      <Alert
                        invert={false}
                        message="Note, you will have the opportunity to grant another household member this preference"
                        subtitle="This application would no longer be eligible for Live Work Preference"
                        title="This change will affect this application's preferences"
                      >
                        <div
                          className="alert-box-and-notice"
                        >
                          <AlertBox
                            closeType="text"
                            invert={false}
                            message="This change will affect this application's preferences"
                            noMargin={true}
                          >
                            <div
                              className="alert-box alert no-margin"
                              data-alert={true}
                            >
                              <span
                                className="alert-icon ui-icon ui-medium"
                              >
                                <svg>
                                  <use
                                    xlinkHref="#i-warning"
                                  />
                                </svg>
                              </span>
                              <p
                                className="alert-body"
                              >
                                This change will affect this application's preferences
                              </p>
                              <button
                                className="button button-link close text"
                                type="button"
                              >
                                Close
                              </button>
                            </div>
                          </AlertBox>
                          <AlertNotice
                            content="Note, you will have the opportunity to grant another household member this preference"
                            invert={false}
                            title="This application would no longer be eligible for Live Work Preference"
                          >
                            <div
                              className="alert-notice alert "
                            >
                              <p
                                className="t-tiny c-alert margin-bottom"
                              >
                                This application would no longer be eligible for Live Work Preference
                              </p>
                              <p
                                className="t-tiny c-steel margin-bottom"
                              >
                                Note, you will have the opportunity to grant another household member this preference
                              </p>
                            </div>
                          </AlertNotice>
                        </div>
                      </Alert>
                    </Component>
                    <Component>
                      <section
                        className="modal-inner"
                      >
                        <div>
                          content
                        </div>
                      </section>
                    </Component>
                    <Component>
                      <footer
                        className="modal-footer bg-dust"
                      >
                        <div
                          className="modal-button-group row"
                        >
                          <div
                            className="modal-button_item modal-button_primary"
                          >
                            <button
                              className="button alert-fill"
                              onClick={[MockFunction]}
                            >
                              update
                            </button>
                          </div>
                          <div
                            className="modal-button_item modal-button_secondary"
                          >
                            <button
                              className="button no-border"
                              onClick={[MockFunction]}
                            >
                              cancel
                            </button>
                          </div>
                        </div>
                      </footer>
                    </Component>
                    <button
                      aria-label="Close modal"
                      className="button button-link close-reveal-modal"
                      onClick={[MockFunction]}
                    >
                      <span
                        className="sr-only"
                      >
                        Close
                      </span>
                      <span
                        className="ui-icon ui-medium i-primary"
                      >
                        <svg>
                          <use
                            xlinkHref="#i-close"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </Component>
              </SimpleModalBody>
            </div>
          </div>
        </ModalPortal>
      </Portal>
    </Modal>
  </Modal>
</SimpleModal>
`;
